### PR TITLE
Alter functions to request sending emails

### PR DESCRIPTION
**Pull request by charlesstrange2**
Add fixtures for this function
Add feedback message
Add appropriate colours or feedback

### CODE REVIEW

File: src/pages/Account.vue [hunk 1]
```diff
@@ -61,16 +63,54 @@
...
+                        var mailMsg = "";
+                        switch (response.status) {
+                            case 202 :
+                                this.goodFeedback = true;
+                                mailMsg = response.data;
+                                break;
+                            default :
+                                this.goodFeedback = false;
+                                mailMsg = "Were sorry, the server responded unusually. Please try again later.";
+                                console.log(response.data); // because we need to see what the server said somewhere.
```
Comment (gingerbreadpup) [inline src/pages/Account.vue]: How about something like "Something went wrong, please try again later."

**Review by gingerbreadpup — APPROVED** [review]

File: src/pages/Account.vue (hunk 1 above)
Comment (charlesstrange2) [reply src/pages/Account.vue]: Agreed. Will alter.

File: src/pages/Account.vue [hunk 2]
```diff
@@ -61,16 +63,54 @@
...
         },
         created: function () {
 
         },
+        methods: {
+            onRequestSubmissionEmail : function(event) {
+                var url = '/traders/' + Store.trader.id + '/voucher-history-email';
+                this.requestEmailBeSent(url,{"submission_date" : event.target.id})
+            },
+            onRequestVoucherHistoryEmail: function() {
```
Comment (katjam) [inline src/pages/Account.vue:84]: Do we necessarily want these to be different? i.e. not just the same but with null date.

**Review by katjam — APPROVED** [review]
Yup. Looks ok to me unless you want to combine history and submission into one method.

:penguin:

File: src/pages/Account.vue [hunk 3]
```diff
@@ -61,16 +63,54 @@
...
+                        // write the response into the page
+                        var mailMsg = "";
+                        switch (response.status) {
+                            case 202 :
+                                this.goodFeedback = true;
+                                mailMsg = response.data;
+                                break;
+                            default :
+                                this.goodFeedback = false;
+                                mailMsg = "Something went wrong, please try again later.";
```
Comment (katjam) [inline src/pages/Account.vue:101]: [xXx] ?

File: src/pages/Account.vue (hunk 2 above)
Comment (charlesstrange2) [reply src/pages/Account.vue:84]: I went for KISS - too lazy to check the id was a date or not.

File: src/pages/Account.vue (hunk 3 above)
Comment (charlesstrange2) [reply src/pages/Account.vue:101]: Steph suggested it - she's Voice of God on design at the mo

**Pull request merged**